src/server/admin/server/api_server.go

```diff
@@ -437,23 +439,32 @@ func (a *apiServer) apply1_8Op(pachClient *client.APIClient, op *admin.Op1_8) er
 		}
 	case op.Branch != nil:
 		if op.Branch.Branch == nil {
-			op.Branch.Branch = client.NewBranch(op.Branch.Head.Repo.Name, op.Branch.SBranch)
+			op.Branch.Branch = client.NewBranch(op.Branch.Head.Repo.Name, ancestry.SanitizeName(op.Branch.SBranch))
 		}
 		if _, err := pachClient.PfsAPIClient.CreateBranch(pachClient.Ctx(), op.Branch); err != nil && !errutil.IsAlreadyExistError(err) {
 			return fmt.Errorf("error creating branch: %v", grpcutil.ScrubGRPC(err))
 		}
 	case op.Pipeline != nil:
-		if op.Pipeline.Salt != "" {
-			// clear salt so we don't re-use old datum hashtrees (which may have an invalid format)
-			op.Pipeline.Salt = ""
-		}
```

Comment:
This was an unrelated changed that I think we wanted. Nulling the salt will cause all datums to be reprocessed which shouldn't be necessary for 1.8 to 1.9 migration, and will cause a lot of unneeded computing to happen.

@msteffen is this change ok? I think you originally added it.

Reply:
I think so. If the datum hashtree format is the same, then this should be totally fine